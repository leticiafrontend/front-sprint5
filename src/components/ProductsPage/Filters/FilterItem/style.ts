import styled, { css } from 'styled-components';

interface PropsFilterItem {
  isSelected: boolean;
}

export const Item = styled.li`
  display: flex;
  align-items: center;
  padding: 12px 14px;
  &:hover {
    cursor: pointer;
    background-color: rgba(38, 30, 30, 0.04);
  }
`;

export const Label = styled.span`
  text-transform: uppercase;
  font-size: 0.6875rem;
  font-weight: 400;
  color: #261e1e;
`;

export const Icon = styled.img<PropsFilterItem>`
  height: 20px;
  margin-left: 8px;

  ${({ isSelected }) =>
    isSelected &&
    css`
      transform: rotate(180deg);
    `}
`;
