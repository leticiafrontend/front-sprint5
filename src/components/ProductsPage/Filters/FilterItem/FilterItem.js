import React from 'react'
import { Icon, Item, Label } from './style'
import filter from '../../../../assets/icons/filter.svg'

export const FilterItem = ({ label }) => (
  <Item>
    <Label>{label}</Label>
    <Icon src={filter} />
  </Item>
)
