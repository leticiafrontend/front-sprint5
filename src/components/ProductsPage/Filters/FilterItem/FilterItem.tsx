import React from 'react';
import { Icon, Item, Label } from './style';
import filter from '../../../../assets/icons/filter.svg';
import { useProducts } from '../../../../hooks/useProducts';

interface PropsFilterItem {
  label: string;
  option: string;
}

export const FilterItem: React.FC<PropsFilterItem> = ({ label, option }) => {
  const { products, setProducts } = useProducts();

  const handleFilter = (option: string) => {
    switch (option) {
      case 'name':
        products.products.sort((a: any, b: any) => {
          return a.name > b.name ? 1 : a.name < b.name ? -1 : 0;
        });
        console.log(products.products);
        break;

      case 'price':
        products.products.sort((a: any, b: any) => {
          return parseInt(a.price) > parseInt(b.price)
            ? 1
            : parseInt(a.price) < parseInt(b.price)
            ? -1
            : 0;
        });
        console.log(products.products);
        break;

      case 'size':
        products.products.sort((a: any, b: any) => {
          return a.sizes < b.sizes ? 1 : a.sizes > b.sizes ? -1 : 0;
        });
        console.log(products.products);
        break;

      case 'color':
        products.products.sort((a: any, b: any) => {
          return a.color > b.color ? 1 : a.color < b.color ? -1 : 0;
        });
        console.log(products.products);
        break;
    }
  };

  return (
    <Item onClick={() => handleFilter(option)}>
      <Label>{label}</Label>
      <Icon src={filter} />
    </Item>
  );
};
