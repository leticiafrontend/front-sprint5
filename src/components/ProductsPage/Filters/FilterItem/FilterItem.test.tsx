import React from 'react'
import { render } from '@testing-library/react'
import { FilterItem } from './FilterItem'

describe('Componente de item do filtro', () => {
  it('O snapshot do componente deve permanecer sempre o mesmo', () => {
    const { container } = render(<FilterItem label='Tamanho' />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
