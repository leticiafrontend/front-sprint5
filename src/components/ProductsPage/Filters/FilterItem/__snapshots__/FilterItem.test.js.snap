// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente de item do filtro O snapshot do componente deve permanecer sempre o mesmo 1`] = `
<li
  class="sc-bdnxRM kYenBY"
>
  <span
    class="sc-gtsrHT bAmSmd"
  >
    Tamanho
  </span>
  <img
    class="sc-dlnjwi ckhPXh"
    src="filter.svg"
  />
</li>
`;
